documentation/dev_running_from_source.markdown

```diff
@@ -14,23 +33,62 @@ Step 0: Quick Start for Developers
     $ git clone --recursive git://github.com/puppetlabs/puppet-server
     $ cd puppet-server
 
-    # Copy sample puppet server config file
-    $ mkdir ~/.puppet-server
-    $ cp dev/puppet-server.conf.sample ~/.puppet-server/puppet-server.conf
-    # Edit the default config file to include your own preferred paths and settings
-    $ vi ~/.puppet-server/puppet-server.conf
-
+    # Remove any old config if you want to make sure you're using the latest
+    # defaults
+    $ rm -rf ~/.puppet-server
     # Copy the sample repl utilities namespace
     $ cp dev/user.clj.sample dev/user.clj
-    # Edit it to suit your needs; in particular, modify the config path to point
-    # to your custom copy of puppet-server.conf
-    $ vi dev/user.clj
 
     # Launch the clojure REPL
     $ lein repl
     # Run Puppet Server
+    user=> (help)
     user=> (go)
 
+You should now have a running server.  Your confdir is set to `./target/master-conf`;
+see ~/.puppet-server/puppet-server.conf to examine this and other default paths.
+
+In another shell, you can run the agent:
+
+    # Go to the directory where you checked out puppet-server
+    $ cd puppet-server
+    # Set ruby and bin paths
+    $ export RUBYLIB="./ruby/puppet/lib:./ruby/facter/lib"
+    $ export PATH=$PATH:./ruby/puppet/bin:./ruby/facter/
+    # Create the modules directory for the production environment, to avoid the
+    #  annoying module error during pluginsync
+    $ mkdir -p ./target/master-code/environments/production/modules
+    # Run the agent
+    $ puppet agent --no-daemonize --debug --trace --verbose \
+         --confdir ./target/master-conf-dir \
+         --vardir=./target/master-var-dir \
+         --server `facter fqdn` --onetime
```

Comment:
Also they can file PRs to change that part if they like since they know more about it than we do, but that ^^ is how I've always run it, and I just tested it.  Works On My Machine (tm)